File: backend/app/schemas/liver_registry.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime, date
from decimal import Decimal
from app.models.liver_registry import LiverDiagnosis, LiverRiskFlag
from app.schemas import PatientOut, UserOut


class LiverRegistryCreate(BaseModel):
    patient_id: int
    facility_id: int
    diagnosis: LiverDiagnosis
    fibroscan_score: Optional[Decimal] = Field(None, ge=0, le=75)
    viral_load: Optional[Decimal] = Field(None, ge=0)
    afp: Optional[Decimal] = Field(None, ge=0)
    alt: Optional[Decimal] = Field(None, ge=0)
    ast: Optional[Decimal] = Field(None, ge=0)
    ultrasound_date: Optional[date] = None
    next_review_date: date


class LiverRegistryUpdate(BaseModel):
    diagnosis: Optional[LiverDiagnosis] = None
    fibroscan_score: Optional[Decimal] = Field(None, ge=0, le=75)
    viral_load: Optional[Decimal] = Field(None, ge=0)
    afp: Optional[Decimal] = Field(None, ge=0)
    alt: Optional[Decimal] = Field(None, ge=0)
    ast: Optional[Decimal] = Field(None, ge=0)
    ultrasound_date: Optional[date] = None
    next_review_date: Optional[date] = None


class LiverRegistryOut(BaseModel):
    id: int
    patient_id: int
    facility_id: int
    recorded_by: int
    diagnosis: LiverDiagnosis
    fibroscan_score: Optional[Decimal] = None
    viral_load: Optional[Decimal] = None
    afp: Optional[Decimal] = None
    alt: Optional[Decimal] = None
    ast: Optional[Decimal] = None
    ultrasound_date: Optional[date] = None
    next_review_date: date
    risk_flag: LiverRiskFlag
    created_at: datetime
    updated_at: Optional[datetime] = None
    patient: Optional[PatientOut] = None
    recorder: Optional[UserOut] = None

    class Config:
        from_attributes = True


class LiverRegistryScanResult(BaseModel):
    scanned_records: int
    overdue_count: int
    trend_alert_count: int
    normal_count: int


class LiverRegistryAlertSummary(BaseModel):
    total_alerts: int
    overdue_count: int
    trend_alert_count: int
    alerts: List[LiverRegistryOut]

File: frontend/types/liver-registry.ts
export type LiverDiagnosis = "HEP_B" | "HEP_C" | "CIRRHOSIS" | "HCC";
export type LiverRiskFlag = "NORMAL" | "OVERDUE" | "TREND_ALERT";

export interface LiverRegistryRecord {
  id: number;
  patient_id: number;
  facility_id: number;
  recorded_by: number;
  diagnosis: LiverDiagnosis;
  fibroscan_score?: number;
  viral_load?: number;
  afp?: number;
  alt?: number;
  ast?: number;
  ultrasound_date?: string;
  next_review_date: string;
  risk_flag: LiverRiskFlag;
  created_at: string;
  updated_at?: string;
  patient?: {
    id: number;
    full_name: string;
    age: number;
    sex: string;
  };
  recorder?: {
    id: number;
    name: string;
    role: string;
  };
}

export interface LiverRegistryAlertSummary {
  total_alerts: number;
  overdue_count: number;
  trend_alert_count: number;
  alerts: LiverRegistryRecord[];
}

export interface LiverRegistryScanResult {
  scanned_records: number;
  overdue_count: number;
  trend_alert_count: number;
  normal_count: number;
}

export const DIAGNOSIS_LABELS: Record<LiverDiagnosis, string> = {
  HEP_B: "Hepatitis B",
  HEP_C: "Hepatitis C",
  CIRRHOSIS: "Liver Cirrhosis",
  HCC: "Hepatocellular Carcinoma",
};

export const RISK_FLAG_LABELS: Record<LiverRiskFlag, string> = {
  NORMAL: "On Track",
  OVERDUE: "Review Overdue",
  TREND_ALERT: "Lab Trend Alert",
};

export const RISK_FLAG_COLORS: Record<LiverRiskFlag, string> = {
  NORMAL: "bg-green-100 text-green-800 border-green-200",
  OVERDUE: "bg-red-100 text-red-800 border-red-200",
  TREND_ALERT: "bg-amber-100 text-amber-800 border-amber-200",
};
